### ISSUE TITLE
C# Proxy ice_getConnection should return an optional.

### ISSUE THREAD

**Issue by externl** [zeroc-ice/ice#2584]
Unless I'm missing something this function can return null.

https://github.com/zeroc-ice/ice/blob/9780db79eb7178648207730c4a1eacd63dc2888a/csharp/src/Ice/Proxy.cs#L442

https://github.com/zeroc-ice/ice/blob/main/csharp/test/Ice/proxy/AllTests.cs#L780-L781

**Comment by pepone** [comment]
Doesn't it return null for collocated tests?

**Comment by externl** [comment]
> Doesn't it return null for collocated tests?

Yes, but we currently return `Connection` instead of `Connection?`.
